### PR TITLE
feat: add connections web-ui url getter

### PR DESCRIPTION
**Pull request by RonenMars**
## Description
Our need:
* if the backend is at api.autokitteh.cloud, automatically set the ui to app.autokitteh.cloud
* if it's anything else at x-api.autokitteh.cloud, set the ui x.autokitteh.cloud
* for local or if the webInterfaceURL is not defined or is an empty string set the default to http://localhost:9982

## Linear Ticket
https://linear.app/autokitteh/issue/UI-707/vscode-point-connections-to-the-webui

## What type of PR is this? (check all applicable)

- [x] 💡 (feat) - A New Feature (non-breaking change which adds functionality)
- [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
- [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
- [ ] 🏎 (perf) - Optimization
- [ ] 📄 (docs) - Documentation - Documentation only changes
- [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
- [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white spaces, formatting, missing semi-colons, etc)
- [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
- [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
- [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

## Code Standards
- [ ] Notify only when user attention is needed, otherwise log to console.
  - [ ] Notifications: short description with context identifiers (e.g. id of the manipulated entity).
  - [ ] Logs: full description with full context identifiers (e.g. deploymentId and the relevant projectId).
- [ ] If you're not sure what is the proper behavior, consult with the product team.
- [ ] Reduce the use of `else` by employing early returns to make code more readable and less nested.
- [ ] MVC Separation: Ensure separation of concerns; views should only be manipulated by controllers (also the computing - sorting, fitlering, etc.), not by services, to maintain a clean MVC architecture.
- [ ] Before implementing custom logic, check if existing functions or utilities (like lodash) offer a simpler solution.
- [ ] Avoid using `!important` in CSS where possible.
- [ ] Use memoization for class calculations.
- [ ] Place constant strings in your code using I18n.

### CODE REVIEW

File: src/constants/api.constants.ts [hunk 1]
```diff
@@ -4,4 +4,28 @@ export const baseURLFromVSCode: string = WorkspaceConfig.getFromWorkspace<string
 export const webUIURLFromVSCode: string = WorkspaceConfig.getFromWorkspace<string>("webInterfaceURL", "");
 
 export const BASE_URL = ValidateURL(baseURLFromVSCode) ? baseURLFromVSCode : "";
-export const WEB_UI_URL = ValidateURL(webUIURLFromVSCode) ? webUIURLFromVSCode : "";
+
+function determineWebUIURL(baseURL: string, configuredWebUIURL: string): string {
+	if (!ValidateURL(baseURL)) {
+		return "http://localhost:9982";
```
Comment (J1za) [inline src/constants/api.constants.ts:10]:
```suggestion
        const LOCAL_FALLBACK_URL = "http://localhost:9982";
	if (!ValidateURL(baseURL)) {
		return LOCAL_FALLBACK_URL;
```

File: src/constants/api.constants.ts [hunk 2]
```diff
@@ -4,4 +4,28 @@ export const baseURLFromVSCode: string = WorkspaceConfig.getFromWorkspace<string
...
+	if (url.hostname.endsWith("-api.autokitteh.cloud")) {
+		const subdomain = url.hostname.split("-api")[0];
+		return `https://${subdomain}.autokitteh.cloud`;
+	}
+
+	if (ValidateURL(configuredWebUIURL)) {
+		return configuredWebUIURL;
+	}
+
+	return "http://localhost:9982";
```
Comment (J1za) [inline src/constants/api.constants.ts:28]:
```suggestion
	return LOCAL_FALLBACK_URL;
```

File: src/constants/api.constants.ts (hunk 1 above)
Comment (RonenMars) [reply src/constants/api.constants.ts]: done

File: src/constants/api.constants.ts (hunk 2 above)
Comment (RonenMars) [reply src/constants/api.constants.ts]: done

**Review by J1za — APPROVED** [review]

**Pull request merged**